sdk/textanalytics/Azure.AI.TextAnalytics.Protocol/README.md

```diff
@@ -12,12 +12,307 @@ At this time, if you are looking to interact with the Azure Cognitive Services T
+```
+
+And you can cast back to a CLR type as you might expect:
+
+```csharp
+dynamic doc = JsonData.FromString(/* a string representing the above document */);
+string name = (string)doc.students[0].name; // name is set to the string "Matt"
+string address = (string) doc.students[1].address[1]; // address is set to the string "Building 34"
+```
+
+##### Conversions to and from POCOs
+
+As we saw earlier, `JsonData` can be constructed from an arbitrary CLR object. When this happens, the object is serialized using `System.Text.Json`. You can also convert a JsonData *to* a CLR object:
+
+```csharp
+class Student
+{
+    string name { get; }
+    string address { get; }
+}
+
+dynamic doc = JsonData.FromString(/* a string representing the above document */);
+Student[] students = doc["students"].To<Student[]>();
+Console.WriteLine(students.Length); // prints 2
+Console.WriteLine(students[0].name); // prints "Matt"
+```
+
+And since `System.Text.Json` is used for the deserialization, you can customize your model classes to give nicer names than what might be in the JSON:
+
+```csharp
+class Student
+{
+    [JsonPropertyName("name")]
+    string Name { get; }
+
+    [JsonPropertyName("address")]
+    string Address { get; }
+}
+
+dynamic doc = JsonData.FromString(/* a string representing the above document */);
+Student[] students = doc["students"].To<Student[]>();
+Console.WriteLine(students.Length); // prints 2
+Console.WriteLine(students[0].Name); // prints "Matt"
+```
+
+### DynamicRequest/DynamicResponse
+
+`DynamicRequest` and `DynamicResponse` are designed for interacting with JSON Based REST APIs. The `Body` property of a `DynamicRequest` and `DyanmicResponse` is a `JsonData` instance. Clients expose a set of *operations* with three methods per operation (for example, for an operation named GetStudents):
+
+| Name                 | Returns                 | Description                                                               |
+|----------------------|-------------------------|---------------------------------------------------------------------------|
+| `GetStudents`        | `DynamicResponse`       | Synchronously invokes the operation and waits for a response.             |
+| `GetStudentsAsync`   | `Task<DynamicResponse>` | Returns a `Task` which represents the eventual response of the operation. |
+| `GetStudentsRequest` | `DynamicRequest`        | Returns a `DynamicRequest` which can be modified and then sent.           |
+
+When using the `<Operation>Request` method, the body of the `DynamicRequest` is initialized to the a JsonData representing the empty object. You can either modify the body directly or set your own version. For example, let's imagine a `CreateStudent` API which let us create a new student, by sending a JSON object representing that student:
+
+```csharp
+DynamicRequest req = client.CreateStudentRequest();
+
+req.Body.Set("name", "Matt");
+var address = req.Body.SetEmptyArray("address");
```

Comment:
I know I proposed it, but now it looks/reads badly. Maybe we should add an enum with sentinel values and allow:
```csharp
req.Body.Set("name", "Matt");
req.Body.Set("address", JsonValue.EmptyArray);
```